### PR TITLE
chore: refactor chain event emits

### PR DESCRIPTION
**Pull request by wemeetagain**
**Motivation**

- small thing noticed when looking thru the code

**Description**

- move event emitting of block and blob sidecar events to the place in code where the rest are emitted
- move blob source metrics with other metrics
- wrap all chain event emits in a callOnNextEventLoop

### CODE REVIEW

File: packages/beacon-node/src/chain/blocks/importBlock.ts [hunk 1]
```diff
@@ -414,6 +386,14 @@ export async function importBlock(
 
   if (this.clock.currentSlot - blockSlot < EVENTSTREAM_EMIT_RECENT_BLOCK_SLOTS) {
     // NOTE: Skip looping if there are no listeners from the API
+
+    if (this.emitter.listenerCount(routes.events.EventType.block)) {
```
Comment (twoeths) [inline packages/beacon-node/src/chain/blocks/importBlock.ts]: when node has connected validators we need to import block asap
need to follow the old implementation to emit all events in the next event loop
```typescript
 // We want to import block asap so call all event handler in the next event loop
  callInNextEventLoop(async () => {
```

Comment (twoeths) [reply packages/beacon-node/src/chain/blocks/importBlock.ts]: at least NetworkProcessor listens for block event to process attestations here https://github.com/ChainSafe/lodestar/blob/42af692453b575621d3ba0ce928c4bea0c6bfbfd/packages/beacon-node/src/network/processor/index.ts#L184

guard the block with `callInNextEventLoop` and we don't have to worry about any new event handlers in the future which could delay the block import

**Comment by nflaig** [comment]
> should all chain events be wrapped in a callOnNextEventLoop ? In practice, it seems that the validator only subscribes to the head event so it shouldn't matter

I noticed Vouch is subscribing to block event but it's not time-sensitive as it is just used for internal block<>slot mapping, so we should be fine to delay it slightly.

**Review by twoeths — APPROVED** [review]

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.21.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.21.0) :tada:

**Pull request merged**